src/plugins/identity/README.md

```diff
@@ -10,10 +10,136 @@ GitHub and Discourse (potentially with a different username on each service).
 We would like to combine these two identities together, so that we can
 represent that user's combined cred properly. The Identity plugin enables this.
 
-Specifically, the instance maintainer can provide a (locally unique) username
-for the user, along with a list of aliases the user is known by, e.g.
-`github/username` and `discourse/other_username`. The aliases are simple string
-representations, that are intended to be easy to maintain by hand in a
-configuration file. Then, the identity plugin will provide a list of
-`NodeContraction`s that can be used by `Graph.contractNodes` to combine the
-user identities as described.
+## Status and Caveats
+
+The Identity plugin currently only applies to new (Weighted)Graphs and Cred
+scores being generated. When including historical Graphs/Cred scores, for
+example to distribute Grain, you may need to contract the identities again.
+This can be tricky to do correctly, as they may have already received Grain
+on their GitHub/Discourse addresses before contraction.
+
+Code-reviewed support for Grain built into SourceCred should handle this in
+the future. But for now, requires the instance maintainer to handle Grain
+account contraction.
+
+## Configuration
+
+Identities can be configured using the `project.json` file.
+By adding entries to the the `identities` array.
+
+As an example of what this would look like:
+
+```js
+[
+  {
+    "type": "sourcecred/project",
+    "version": "0.4.0",
+  },
+  {
+    // The identities we should collapse.
+    "identities": [
+      {
+        "username": "User-A",
+        "aliases": ["github/user-a", "discourse/user-a"],
+      },
+      {
+        "username": "User-B",
+        "aliases": ["github/user-b", "discourse/user-b"],
+      },
+    ],
+
+    "id": "@example-project",
+    "discourseServer": {
+      // ...
+    },
+    "repoIds": [
+      // ...
+    ],
+  },
+]
+```
+
+### Username
+
+The `username` field allows you to choose a new username for the collapsed
+identity.
+
+It's case-sensitive, must be _unique in this project_ and can contain `A-Z`,
+`a-z`, `0-9`, `-` or `_` as valid characters.
+
+For display purposes the username is prefixed with an `@`.
+If a username is prefixed with `@` in the project file, it will be ignored.
+Meaning `"username": "User-A"` and `"username": "@User-A"` are equivalent.
+
+### Aliases
+
+Aliases are strings with a format of: `{plugin}/{user ID for this plugin}`
+Currently supported plugins are:
+
+| Plugin    | Example alias            | Notes                                                              |
+|:----------|:-------------------------|:-------------------------------------------------------------------|
+| GitHub    | `github/example-user`    | A user account on GitHub's platform.                               |
+| Discourse | `discourse/example-user` | A user account on the Discourse server configured in this project. |
+
+## How a contraction works
```

Comment:
Took at stab at rewording the below section for flow.

Suggested: 

## Combining Entities

Below is a simplified illustration of how combining aliases works under the hood. 

Note that combining of aliases is performed after the plugins have created their respective part of the graph, but before any cred scores are calculated. 

Let’s assume we have a user, `abc`, who has both GitHub and Discourse accounts which we want to combine.

Running the GitHub plugin creates an identity node with two edges: a Pull Request (PR) and an Issue.

```
GitHub user @abc
├── authored Pull Request #123
└── authored Issue #456
```

Running the Discourse plugin creates an identity node with two edges, both posts `abc` has authored. 

```
Discourse user /u/abc
├── authored Post /t/123/1
└── authored Post /t/123/2
```

Note that the GitHub and Discourse plugin have created separate graphs, which have not yet been merged. 

First, the identity plugin creates a new identity node, with the username provided in the `project.json` file, while removing the alias nodes.

```
Identity @abc

[REMOVED]
├── authored Pull Request #123
└── authored Issue #456

[REMOVED]
├── authored Post /t/123/1
└── authored Post /t/123/2
```

Then the edges are connected to the new identity node. 

```
Identity @abc
This conversation was marked as resolved by vsoch
├── authored Pull Request #123
├── authored Issue #456
├── authored Post /t/123/1
└── authored Post /t/123/2
```

The SourceCred core algorithm can then next calculate cred scores.

Reply:
> Running the GitHub plugin creates an identity node with ...
> Running the Discourse plugin creates an identity node with ...

They do not create identity nodes. They create GitHub user nodes and Discourse user nodes resp. The identity plugin creates identity nodes.

However all 3 of these nodes are marked as being "user node" types. Which we treat differently from other nodes in a few cases. For example, we would distribute Grain to user node types, rather than every node in the graph. `scores.json` files also filter to only include user node types.

---

> Note that the GitHub and Discourse plugin have created separate graphs, which have not yet been merged.

This seems a bit misleading to me in both terminology and visualization. Technically at this point we've already merged data from the different plugins into a single graph (why I visualized it as a single code block). However these sections of the graph are disconnected from each other, because they don't have any edges between them. After the contraction, they would be connected because they share the same author.

This disconnect while being in the same graph is relevant for seeing why Cred _flow_ also changes when contracting identities. For example, in the disconnected situation, Cred you've earned through contributions on GitHub, would not impact how much Cred can flow through your Discourse likes. While in the connected situation this "path" in the graph now exists and Cred can flow between services.

I'll elaborate more on this in the example to point this out as I explain the contraction itself.